smart_contract/test/Faucet.test.ts

```diff
@@ -0,0 +1,255 @@
+
+      const withdrawalAmount = await faucet.withdrawalAmount()
+      await deployer.sendTransaction({ to: faucetAddress, value: withdrawalAmount.toString() })
+
+      const currentDeployerBalance = await deployer.provider.getBalance(deployerAddress)
+
+      await faucet.connect(deployer).withdraw()
+
+      await expect(await deployer.provider.getBalance(deployerAddress)).to.be.greaterThanOrEqual(currentDeployerBalance)
+    })
+
+    it('Faucet admin can add new minter', async function () {
+      const [, wallet2] = await ethers.getSigners()
+      const wallet2Address = await wallet2.getAddress()
+
+      const { faucet, deployer } = await loadFixture(deploy)
+
+      await faucet.connect(deployer).addMinter(wallet2Address)
+
+      await expect(await faucet.isMinter(wallet2Address)).to.be.true
+    })
+
+    it('Faucet admin can remove minter', async function () {
+      const [, wallet2] = await ethers.getSigners()
+      const wallet2Address = await wallet2.getAddress()
+
+      const { faucet, deployer } = await loadFixture(deploy)
+
+      await faucet.connect(deployer).addMinter(wallet2Address)
+
+      await expect(await faucet.isMinter(wallet2Address)).to.be.true
+
+      await faucet.connect(deployer).removeMinter(wallet2Address)
+
+      await expect(await faucet.isMinter(wallet2Address)).to.be.false
+    })
+
+    it('Faucet admin can add new admin', async function () {
+      const [, wallet2] = await ethers.getSigners()
+      const wallet2Address = await wallet2.getAddress()
+
+      const { faucet, deployer } = await loadFixture(deploy)
+
+      await faucet.connect(deployer).addAdmin(wallet2Address)
+
+      await expect(await faucet.isAdmin(wallet2Address)).to.be.true
+    })
+
+    it('Faucet admin can remove admin', async function () {
+      const [, wallet2] = await ethers.getSigners()
+      const wallet2Address = await wallet2.getAddress()
+
+      const { faucet, deployer } = await loadFixture(deploy)
+
+      await faucet.connect(deployer).addAdmin(wallet2Address)
+
+      await expect(await faucet.isAdmin(wallet2Address)).to.be.true
+
+      await faucet.connect(deployer).removeAdmin(wallet2Address)
+
+      await expect(await faucet.isAdmin(wallet2Address)).to.be.false
+    })
+  })
+
+  describe('Request tokens by non-minter role', function () {
+    it('Non-Minter cannot dispatch native token to wallet2', async function () {
+      const [, wallet2] = await ethers.getSigners()
+      const wallet2Address = await wallet2.getAddress()
+      const currentWallet2Balance = await wallet2.provider.getBalance(wallet2Address)
+
+      const { faucet, faucetAddress, deployer } = await loadFixture(deploy)
+
+      const withdrawalAmount = await faucet.withdrawalAmount()
+      await deployer.sendTransaction({ to: faucetAddress, value: withdrawalAmount.toString() })
+
+      await expect(faucet.connect(wallet2).requestTokens(wallet2Address)).to.be.revertedWith(
+        errors.MinterRole.NotMinter,
```

Comment:
This test is failing because message must have changed. However, when we update to error types this logic will change, so maybe this doesn't really matter. All of the revert tests are failing because of messages.

Reply:
Yes I changed for .revertedWithCustomError